shared/src/native-src/loader.cpp

```diff
@@ -1391,6 +1536,125 @@ namespace shared
+            // https://github.com/dotnet/runtime/issues/11878#issuecomment-457149220
+            if ((moduleFlags & COR_PRF_MODULE_FLAT_LAYOUT) != 0)
+            {
+                Debug("Loader::GetModuleEntryPointToken: Module with flat layout detected.");
+                IMAGE_SECTION_HEADER* sectionRet = nullptr;
+                const auto pSection = pHeaders + FIELD_OFFSET(IMAGE_NT_HEADERS, OptionalHeader) +
+                                      VAL16(ntHeaders->FileHeader.SizeOfOptionalHeader);
+
+                Debug("Loader::GetModuleEntryPointToken: Number of sections: " +
+                      ToString(VAL16(ntHeaders->FileHeader.NumberOfSections)));
+
+                // We need to iterate and find the right section
+                auto section = (IMAGE_SECTION_HEADER*) pSection;
+                for (auto i = 0; i < VAL16(ntHeaders->FileHeader.NumberOfSections); i++)
+                {
+                    // https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#section-data
+                    const auto sectionVirtualAddress = VAL32(section->VirtualAddress);
+                    const auto sectionVirtualSize = (UINT) VAL32(section->Misc.VirtualSize);
+
+                    // The alignment (in bytes) of sections when they are loaded into memory.
+                    // It must be greater than or equal to FileAlignment. The default is the page size for the
+                    // architecture.
+                    const auto sectionAlignment = (UINT) VAL32(ntHeaders->OptionalHeader.SectionAlignment);
+
+                    const auto sectionAlignUpValue =
+                        (sectionVirtualSize + sectionAlignment - 1) & ~(sectionAlignment - 1);
+
+                    if (clrDataVirtualAddress < sectionVirtualAddress + sectionAlignUpValue)
+                    {
+                        if (clrDataVirtualAddress < sectionVirtualAddress)
+                        {
+                            Debug("Loader::GetModuleEntryPointToken: " + ToString(i) + ". sectionRet = nullptr");
+                            sectionRet = nullptr;
+                        }
+                        else
+                        {
+                            Debug("Loader::GetModuleEntryPointToken: " + ToString(i) + ". sectionRet = (found)");
+                            sectionRet = section;
+                            break;
+                        }
+                    }
+
+                    // Advance to the next section.
+                    section++;
+                }
+
+                if (sectionRet == nullptr)
+                {
+                    pCorHeader = moduleBaseLoadAddress + clrDataVirtualAddress;
+                }
+                else
+                {
+                    pCorHeader = moduleBaseLoadAddress + clrDataVirtualAddress - VAL32(sectionRet->VirtualAddress) +
+                                 VAL32(sectionRet->PointerToRawData);
+                }
```

Comment:
can you put a comment to explain where it comes ?

Reply:
Already put the comment here: https://github.com/DataDog/dd-trace-dotnet/blob/ec19367992de484cbf41c0e8e57e7c86e10e4eb8/shared/src/native-src/loader.cpp#L1572

In the comment appears that formula: 
> BySection means you first parse the PE Section table, and find the section where PointerToRawData <= X < PointerToRawData + SizeOfRawData. Then byte X is placed in memory at baseLoadAddress + VirtualAddress + (X - PointerToRawData).
>When you are reading memory at baseLoadAddress + directoryEntry.VirtualAddress, that is the correct place to read if the file has been mapped into memory BySection. However if it was mapped Flat then you need to parse the section table and do a little math to figure out the correct address to read from.

Reply:
oh ok sorry. i should have check the link first.